src/gui/qgscredentialdialog.cpp

```diff
@@ -45,7 +56,58 @@ QgsCredentialDialog::QgsCredentialDialog( QWidget *parent, Qt::WindowFlags fl )
   connect( this, &QgsCredentialDialog::credentialsRequestedMasterPassword,
            this, &QgsCredentialDialog::requestCredentialsMasterPassword,
            Qt::BlockingQueuedConnection );
-  mOkButton = buttonBox->button( QDialogButtonBox::Ok );
+
+  // Setup ignore button
+  mIgnoreButton->setToolTip( tr( "All requests for this connection will be automatically rejected" ) );
+  QMenu *menu = new QMenu( mIgnoreButton );
+  QAction *ignoreTemporarily = new QAction( tr( "Ignore for 10 Seconds" ), menu );
+  ignoreTemporarily->setToolTip( tr( "All requests for this connection will be automatically rejected for 10 seconds" ) );
+  QAction *ignoreForSession = new QAction( tr( "Ignore for Session" ), menu );
+  ignoreForSession->setToolTip( tr( "All requests for this connection will be automatically rejected for the duration of the current session" ) );
+  menu->addAction( ignoreTemporarily );
+  menu->addAction( ignoreForSession );
+  connect( ignoreTemporarily, &QAction::triggered, [ = ]
+  {
+    mIgnoreMode = IgnoreTemporarily;
+    //mIgnoreButton->setText( ignoreTemporarily->text() );
```

Comment:
why is this commented? I think it needs to be there -- currently if I change the mode the button doesn't reflect the new value.

Reply:
@elpaso did you see this note?

Reply:
No, sorry, I missed this. Looking at it now.